bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/aWATTarBestpriceHandler.java

```diff
@@ -0,0 +1,287 @@
+                if (res2.getPriceSum() < res.getPriceSum()) {
+                    res = res2;
+                }
+            }
+            result = res;
+        } else {
+            List<aWATTarPrice> range = getPriceRange(bridgeHandler, timerange,
+                    (o1, o2) -> Double.compare(o1.getPrice(), o2.getPrice()));
+            aWATTarNonConsecutiveBestPriceResult res = new aWATTarNonConsecutiveBestPriceResult(config.length,
+                    bridgeHandler.getTimeZone());
+            logger.trace("Bestprice Candidate range: {}", range);
+            int ct = 0;
+            for (aWATTarPrice price : range) {
+                res.addMember(price);
+                if (++ct >= config.length) {
+                    break;
+                }
+            }
+            result = res;
+        }
+        logger.trace("refreshChannel {}: result is: {}", channelUID, result);
+        String channelId = channelUID.getIdWithoutGroup();
+        long diff;
+        switch (channelId) {
+            case CHANNEL_ACTIVE:
+                state = OnOffType.from(result.isActive());
+                break;
+            case CHANNEL_START:
+                state = getDateTimeType(result.getStart(), timeZoneProvider);
+                break;
+            case CHANNEL_END:
+                state = getDateTimeType(result.getEnd(), timeZoneProvider);
+                break;
+            case CHANNEL_COUNTDOWN:
+                diff = result.getStart() - Instant.now().toEpochMilli();
+                if (diff >= 0) {
+                    state = new StringType(getDuration(diff));
+                }
+                break;
+            case CHANNEL_REMAINING:
+                diff = result.getEnd() - Instant.now().toEpochMilli();
+                if (result.isActive()) {
+                    state = new StringType(getDuration(diff));
+                }
+                break;
+            case CHANNEL_HOURS:
+                state = new StringType(result.getHours());
+                break;
+            default:
+                logger.warn("Unknown channel id {} for Thing type {}", channelUID, getThing().getThingTypeUID());
+        }
+        updateState(channelUID, state);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Handling command {} for channel {}", command, channelUID);
+        if (command instanceof RefreshType) {
+            refreshChannel(channelUID);
+        } else {
+            logger.debug("Binding {} only supports refresh command", BINDING_ID);
+        }
+    }
+
+    private List<aWATTarPrice> getPriceRange(aWATTarBridgeHandler bridgeHandler, Timerange range,
+            Comparator<aWATTarPrice> comparator) {
+        ArrayList<aWATTarPrice> result = new ArrayList<>();
+        SortedMap<Long, aWATTarPrice> priceMap = bridgeHandler.getPriceMap();
+        if (priceMap == null) {
+            logger.error("No prices available, returning empty result");
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be debug. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
See above: I removed most of the logging statements.